python/taichi/lang/matrix_ops_utils.py

```diff
@@ -75,19 +56,44 @@ def assert_tensor(m, msg='not tensor type: {}'):
     raise TaichiCompilationError(msg.format(type(m)))
 
 
-# TODO(zhanlue): rearrange to more generic checker functions
-# for example: "assert_is_instance(args, indices=[], instances=[], logic='or')"
 def assert_vector(v, msg='not a vector: {}'):
     if (isinstance(v, Expr) or isinstance(v, Matrix)) and len(
             v.get_shape()) == 1:
         return True, None
-    raise TaichiCompilationError(msg.format(type(v)))
+    return False, msg.format(type(v))
 
 
 def assert_list(x, msg='not a list: {}'):
     if isinstance(x, list):
         return True, None
-    raise TaichiCompilationError(msg.format(type(x)))
+    return False, msg.format(type(x))
+
+
+def arg_foreach_check(*arg_indices, fns=[], logic='or', msg=None):
+    def check(*args, **kwargs):
+        for i in arg_indices:
```

Comment:
It seems that the logic below cannot work with multiple indices.

Reply:
Ah I'm saying the problem logic, not the `logic` argument :-) It seems that the function will return anyway after the first iteration, so if `arg_indices` contains multiple indices it may not work properly.